lib/auth/oauth2.js

```diff
@@ -69,6 +90,7 @@ const oauth2 = {
    * @param {string} [code] - An authorization code. Deprecated since 3.16.0, use other helper methods instead.
    * @param {object} [params] - Optional parameters such as `redirect_uri`.
    * @returns {string}
+   * @deprecated since 3.16.0, use other helper methods instead.
```

Comment:
```suggestion
   * @deprecated since 4.0.0, use other helper methods instead.
```
?

Reply:
The code arg was deprecated on 3.16.0 (see jsdoc), so let's deprecate the whole method also from the same version.